components/playlist/browser/playlist_download_request_manager.cc

```diff
@@ -95,16 +104,24 @@ void PlaylistDownloadRequestManager::FetchPendingRequest() {
 }
 
 void PlaylistDownloadRequestManager::RunMediaDetector(Request request) {
+  DVLOG(2) << __func__;
   CHECK(PlaylistJavaScriptWorldIdIsSet());
 
   DCHECK_GE(in_progress_urls_count_, 0);
   in_progress_urls_count_++;
 
   DCHECK(callback_for_current_request_.is_null());
   callback_for_current_request_ = std::move(request.callback);
+  DCHECK(callback_for_current_request_)
+      << "Empty callback shouldn't be requested";
+  request_start_time_ = base::Time::Now();
 
   if (absl::holds_alternative<std::string>(request.url_or_contents)) {
+    // Start to request on clean slate, so that result won't be affected by
+    // previous page.
+    web_contents_.reset();
```

Comment:
This reset needs here? Can we add `DCHECK(!web_contents);` at the start of `CreateWebContents()` instead?

Reply:
Yes, that'd be better 👍

Reply:
Ah, I mean adding `DCHECK()` to the start of `CreateWebContents();` :)

Reply:
Oh, yeah. That's even better.